modal/_utils/bytes_io_segment_payload.py

```diff
@@ -52,12 +52,16 @@ def reset_state(self):
         self._value.seek(self.initial_seek_pos)
 
     @contextmanager
-    def reset_on_error(self):
+    def reset_on_error(self, subtract_progress: bool = False):
```

Comment:
Do we need intra-block progress?

Reply:
In my opinion, it would be good to keep using the existing `BytesIOSegmentPayload` instead of trying to create a new thing.